lib/global_uid/base.rb

```diff
@@ -59,22 +59,22 @@ def self.with_servers
         servers.shuffle! # subsequent requests are made against different servers
       end
 
-      raise NoServersAvailableException if servers.empty?
-
       errors = []
       servers.each do |server|
         begin
           yield server if server.active?
         rescue TimeoutException, Exception => e
-          notify e, "#{e.message}"
+          notify(e, e.message)
           errors << e
           server.disconnect!
           server.update_retry_at(1.minute)
         end
       end
 
       if connections.empty?
```

Comment:
`servers.each` followed by `connections.empty?` is a bit confusing to read. Do I understand correctly that `connections` are the non-disconnected mysql2_connections from the `servers` array?

Reply:
Maybe what I’m saying is, should we rename the method to `#active_connections`?

Reply:
Agreed, thanks. Good suggestion.